addons/website/models/website.py

```diff
@@ -1719,9 +1733,15 @@ def get_website_page_ids(self):
         return pages_by_website
 
     def _get_website_pages(self, domain=None, order='name', limit=None):
+        # ==== User Defined Redirections ====
+        redirects = self.env["website.rewrite"].sudo().search([])
+        redirect_map = {r.url_from: r.url_to for r in redirects}
         website = self.get_current_website()
         domain = Domain(domain or Domain.TRUE) & website.website_domain()
         pages = self.env['website.page'].sudo().search(domain, order=order, limit=limit)
+        for page in pages:
+            if page.url in redirect_map:
+                page.url = redirect_map[page.url]
```

Comment:
```suggestion
        for page in pages:
            redirect_url = redirect_map.get(page.url)
            if redirect_url:
                page.url = redirect_url
```
isn't this version will be more better?